astropy/io/misc/yaml.py

```diff
@@ -243,6 +243,28 @@ class AstropyDumper(yaml.SafeDumper):
     def _represent_tuple(self, data):
         return self.represent_sequence("tag:yaml.org,2002:python/tuple", data)
 
+    def represent_float(self, data):
+        # Override to change repr(data) to str(data) since otherwise all the
+        # numpy scalars fail in not NUMPY_LT_1_20.
+        if data != data or (data == 0.0 and data == 1.0):
```

Comment:
Just to make sure I understand, why not np.isnan?

Reply:
I assume that's for performance reasons, though, that's a question for @mhvk 
(`math.isnan` should also be faster than `np.isnan` for scalars)

Reply:
Hmm, I vaguely recall that I just copied the snipped from how pyyaml does it, except changing `repr(data)` to `str(data)` (I *clearly* should have added a note with that...). I'm not quite sure why they worked this way - perhaps there are python float-like's that do not support `isnan`?

Reply:
Ok no problem - if we can get things to work with ``math.isnan`` I think that would be preferable

Reply:
For the record here's the line you apparently copied https://github.com/yaml/pyyaml/blob/48838a3c768e3d1bcab44197d800145cfd0719d6/lib/yaml/representer.py#L172
It was written 15 years ago (circa 2009), while `math.isnan` was introduced in Python 2.6, released in October 2008, so it's possible that the only reason they didn't use it was backward compatibility, and it just never got refactored.
Since the whole function is copied from pyyaml, it seems best to keep it as is, but I'll add a note so this knowledge doesn't get lost.

Reply:
FYI @mhvk I also updated your comment. It said that numpy scalars failed on "not NUMPY_LT_1_20" which I assume was a mistake and you really meant "not NUMPY_LT_2_0"

Reply:
Thanks, @neutrinoceros, that all makes sense.